miqi/runtime/tool_registry_factory.py

```
@@ -46,6 +46,69 @@ def _default_read_roots() -> list[Path]:
+            return "deny"
+        choice_id = str((gate_result.get("answers") or {}).get("choice_id") or "")
+        if choice_id == "allow_always":
+            # 统一入口：runtime + config 原子成对（外部审阅 #854）
+            if sandbox_manager is not None:
+                sandbox_manager.allow_system_installs = True
+            try:
+                config.tools.sandbox.allow_system_installs = True
+                from miqi.config.loader import save_config
+
+                save_config(config)
+            except Exception as exc:  # noqa: BLE001 - runtime 已生效，持久化失败仅告警
```

Ruff (0.16.2)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)